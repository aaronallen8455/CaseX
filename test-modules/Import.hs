module Import
  ( Foo(..)
  ) where

data Foo = Foo | Bar | Baz
