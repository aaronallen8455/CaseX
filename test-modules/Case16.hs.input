{-# LANGUAGE OrPatterns #-}
import           Import

main :: IO ()
main = pure ()

test :: Maybe (Maybe Foo) -> Int
test x = case x of
  Just (Nothing; SPLIT) -> 1
  Nothing -> 2
  Just (Just (Foo; SPLIT)) -> 3
