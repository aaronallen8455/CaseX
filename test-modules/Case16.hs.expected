{-# LANGUAGE OrPatterns #-}
import           Import

main :: IO ()
main = pure ()

test :: Maybe (Maybe Foo) -> Int
test x = case x of
  Just (Nothing) -> 1
  Just (Just Bar) -> 1
  Just (Just Baz) -> 1
  Nothing -> 2
  Just (Just (Foo)) -> 3
